Rename 'environment' config option & add singletons setup (#96)
OpenZeppelin/openzeppelin-test-helpers: README.md

### Configuration

> This library features support for both web3 and truffle contract instances: functions like `expectEvent` can receive receipts from either type of contract abstraction.

Reviewer: ```suggestion
This library supports both web3 and truffle contract instances. Where possible, helpers will automatically detect what you're using and work with both. For details about each helper see the documentation below.
```

> In a `truffle` environment, the web3 provider will be pulled from truffle's global web3 instance, otherwise, it defaults to `http://localhost:8545`.

Reviewer: `truffle` was formatted as code here because it was a reference to the `environment` option value.

```suggestion
In a truffle environment, the web3 provider will be pulled from truffle's global web3 instance. Otherwise, it defaults to `http://localhost:8545`.
```

Reviewer: This is a bit of a nitpick but it's part of a bigger issue, which is that after "removing" the environment option I think the readme was left in a frankestein state where things around environment and contract abstraction are not very clear.

Author: Yes, it did feel weird while editing it. Tbh I think we should rethink that whole section.

Reviewer: Yeah I agree. If you have time do you want to give it a shot?

### Configuration

This library supports both web3 and truffle contract instances. Where possible, helpers will automatically detect what you're using and work with both. For details about each helper see their documentation entries below.

> #### provider

Reviewer: ```suggestion
#### `provider`
```

In a truffle environment, the web3 provider will be pulled from truffle's global web3 instance. Otherwise, it defaults to `http://localhost:8545`. You can override this behavior and configure your own via the `provider` key:

```javascript
require('@openzeppelin/test-helpers/configure')({ provider: 'http://localhost:8080' });
````

> #### singletons

Reviewer: ```suggestion
#### `singletons`
```

### Configuration

This library supports both web3 and truffle contract instances. Where possible, helpers will automatically detect what you're using and work with both. For details about each helper see their documentation entries below.

#### provider

In a truffle environment, the web3 provider will be pulled from truffle's global web3 instance. Otherwise, it defaults to `http://localhost:8545`. You can override this behavior and configure your own via the `provider` key:

```javascript
require('@openzeppelin/test-helpers/configure')({ provider: 'http://localhost:8080' });
````

#### singletons

The `singletons` helper returns contract objects, which have multiple values that can be configured:
 * `abstraction`: the underlying contract abstraction type, `'web3'` for `web3-eth-contract` and `'truffle'` for `@truffle/contract` instances. Defaults to `'web3'` unless a truffle environment is detected.

>  * `defaultGas`: how much gas to allocate when a transaction's `gas` field is not specified. Defaults to 8 million.

Reviewer: The 8 million default is waaay too high. Absolutely too high. Dangerously too high. Even if test-helpers is meant to run in testing environments we don't know if someone will use it in mainnet.

Besides, this will probably not work in any non-Ganache blockchain because it's so close or above the block gas limit.

Author: I could've sworn that ganache's block limit was 8 million, but turns out it is instead 6721975. What default value do you suggest?
